src/names.jl

```diff
@@ -152,3 +152,37 @@ function named_axiskeys(A::AbstractArray)
     NT = NamedTuple{dimnames(A)}
     NT(axiskeys(A))
 end
+
+
+"""
+    rekey(A, (1:10, [:a, :b]))
+    rekey(A, 2 => [:a, :b])
+    rekey(A, :y => [:a, :b])
+    rekey(A, :y => q => [:a, :b])
+
+Rekey a KeyedArray via `Tuple`s, `dim => newkey`. If `A` also has named dimensions then you
+can also pass `dimname => newkey`, or even `oldname => newname => newkey` to both `rename`
+and `rekey` the specified dimension.
```

Comment:
Is rekeying and renaming at some time a common operation?
Mightn't it be clearer to seperate it into two operations?

I would feel more comfortable with this PR if it only rekeyed.
And then we could do a follow up PR with the rekey + rename, that could be reveiwed can considered seperately

Reply:
That's a good point. Also, should something like `rename` be defined for NamedDimsArrays first?

Reply:
I believe there is a `rename`  in NamedDims.jl

Reply:
Are we still okay with the pairs syntax or do we want to use kwargs?

Reply:
Okay, I've just dropped the rekey and rename method for now. I think that kind of functionality is only common in the DataFrames ecosystem (`combine`).